homeassistant/components/amcrest/camera.py

```diff
@@ -536,115 +538,123 @@ def _change_setting(
         # is_streaming and is_recording, we can't leave
         # video stream off if recording is being turned on.
         if not self.is_streaming and enable:
-            self._enable_video(True)
-        self._change_setting(enable, "recording", "_is_recording")
+            await self._async_enable_video(True)
+        await self._async_change_setting(enable, "recording", "_is_recording")
 
-    def _get_motion_detection(self) -> bool:
-        return self._api.is_motion_detector_on()
+    async def _async_get_motion_detection(self) -> bool:
+        return await self._api.async_is_motion_detector_on()
 
-    def _set_motion_detection(self, enable: bool) -> None:
+    async def _async_set_motion_detection(self, enable: bool) -> None:
         # The property has a str type, but setter has bool type, which causes mypy confusion
-        self._api.motion_detection = enable  # type: ignore[assignment]
+        await self._api.async_set_motion_detection(enable)
 
-    def _enable_motion_detection(self, enable: bool) -> None:
+    async def _async_enable_motion_detection(self, enable: bool) -> None:
         """Enable or disable motion detection."""
-        self._change_setting(enable, "motion detection", "_motion_detection_enabled")
+        await self._async_change_setting(
+            enable, "motion detection", "_motion_detection_enabled"
+        )
 
-    def _get_audio(self) -> bool:
-        return self._api.audio_enabled
+    async def _async_get_audio(self) -> bool:
+        return await self._api.async_audio_enabled
 
-    def _set_audio(self, enable: bool) -> None:
-        self._api.audio_enabled = enable
+    async def _async_set_audio(self, enable: bool) -> None:
+        await self._api.async_set_audio_enabled(enable)
 
-    def _enable_audio(self, enable: bool) -> None:
+    async def _async_enable_audio(self, enable: bool) -> None:
         """Enable or disable audio stream."""
-        self._change_setting(enable, "audio", "_audio_enabled")
+        await self._async_change_setting(enable, "audio", "_audio_enabled")
         if self._control_light:
-            self._change_light()
+            await self._async_change_light()
 
-    def _get_indicator_light(self) -> bool:
+    async def _async_get_indicator_light(self) -> bool:
         return (
             "true"
-            in self._api.command(
-                "configManager.cgi?action=getConfig&name=LightGlobal"
+            in (
+                await self._api.async_command(
+                    "configManager.cgi?action=getConfig&name=LightGlobal"
+                )
             ).content.decode()
         )
 
-    def _set_indicator_light(self, enable: bool) -> None:
-        self._api.command(
+    async def _async_set_indicator_light(self, enable: bool) -> None:
+        await self._api.async_command(
             f"configManager.cgi?action=setConfig&LightGlobal[0].Enable={str(enable).lower()}"
         )
 
```

Comment:
Existing, but this should be moved to the pypi code in a future PR

Reply:
Yes, I agree, I don't have a doorbell or other camera with a light to test this out to move it over, otherwise may have tried already. The light control and the privacy mask control (in switch) both should be moved upstream.